packages/common/src/ContractUtils.ts

```diff
@@ -88,6 +88,64 @@ export type EventSearchOptions = {
   // filters all events where "myNumber" is 12 or 13.
 };
 
+/**
+ * Return all events between block range. Will paginate the event search using the `pageSize` if specified. Takes as
+ * input a list of async callbacks that are used to make event requests. This means that different event types and
+ * and contracts can be queried in this function.
+ * @param contractEventQueryCallbacks Array of functions that take in `EventSearchOptions` and return promises that will
+ * resolve to arrays of EventData.
+ * Example of one such callback: (blockSearchConfig) => return this.oracle.getPastEvents("RequestPrice", blockSearchConfig)
+ * @param earliestBlockToQuery First block to query.
+ * @param latestBlockToQuery Latest block to query.
+ * @param pageSize Number of blocks to search for in each query. Determines how many web3 requests are sent to fetch
+ * data for all blocks between `earliestBlockToQuery` and `latestBlockToQuery`.
+ * @return array of event data. Array size is equal to size of `contractEventQueryCallbacks`. Also returns number of
+ * web3 requests sent.
+ */
+export async function getEventsWithPaginatedBlockSearch(
+  contractEventQueryCallbacks: ((blockSearchConfig: EventSearchOptions) => Promise<EventData[]>)[],
```

Comment:
Just OOC, why did you decide to pass in an array of events to trigger rather than having the caller just call this function `n` times within a `Promise.all`?

Reply:
Oh that's a good point I hadn't thought of that, I'll refactor as this is more sustainable